plutus-core/plutus-core/src/PlutusCore/Builtin/HasConstant.hs

```diff
@@ -29,6 +32,16 @@ class HasConstant term where
 -- and connects @term@ and its @uni@.
 type HasConstantIn uni term = (UniOf term ~ uni, HasConstant term)
 
+-- | Wrap a Haskell value (given its explicit type tag) as a @term@.
+fromValueOf :: HasConstant term => UniOf term (Esc a) -> a -> term
```

Comment:
> Any reason not to make this actually take a `ValueOf`?

Unnecessary wordiness at the call site where we'd have to make it a composition of the two.

> Otherwise the name is a bit confusing.

`ValueOf` (the constructor) takes a type tag and a value, which is how it got its name. This function also take a type tag and a value, which is how it got its `ValueOf` suffix, and embeds them somewhere. which is how it got its `from` prefix.

Given that there's `fromValue` right below, I think it's pretty unambiguous that `fromValueOf` doesn't take a `ValueOf`, it just got its suffix the same way `ValueOf` got its name.

> Generally for this kind of conversion I wonder if it's useful to mention the target type too, otherwise it's a bit confusing since we have conversions in both directions exported from this module. 

We're literally polymorphic over the target type in `fromValueOf`. We use `term`, because we do occasionally need it for terms, but that `term` most of the time is the value type of some evaluator (like `CkValue` or `CekValue`), so I don't think we should hardcore the `term` part into the name.

I do generally believe it's fine to omit the prefix when you're polymorphic over the type of the argument and omit the suffix when you're polymorphic over the type of the result.